frame/executive/src/lib.rs

```diff
@@ -562,6 +589,24 @@ mod tests {
 		t.into()
 	}
 
+	#[test]
+	fn current_extrinsic_data_is_available() {
+		new_test_ext(1).execute_with(|| {
+			let xt = TestXt::new(Call::Custom(custom::Call::some_function()), sign_extra(1, 0, 0));
+			let xt_hash = <Runtime as frame_system::Trait>::Hashing::hash(&xt.encode());
+			assert!(Executive::apply_extrinsic(xt).is_ok());
+			let eventrecs = System::events();
+            let found_saw_hash_event = eventrecs.iter().any(|rec| {
+                match rec.event {
+                    MetaEvent::custom(custom::Event::<Runtime>::SawExtHash(h)) => h == xt_hash,
+					_ => false
+				}
+			});
+
+			assert!(found_saw_hash_event)
```

Comment:
I think reading a mock event is a very long way to check this. You can just read directly from storage at this point and check that correct hash is placed in storage after `Executive::apply_extrinsic(xt)`

Reply:
Closed this but I wanted to point out I think this is actually a stronger test exactly because it goes a sort of roundabout way - it's demonstrating that the new interfaces I wrote for fetching the extrinsic data are available regardless of how they got there (ie: That it specifically happens because of apply_extrinsic). So the test should remain (relatively) stable in the face of internal changes.

Hopefully that's useful background.